### PR TITLE
Custom redirect after login

### PR DESCRIPTION
**Pull request by bastianallgeier**
When a user logs in that does not have access to the site view, they end up on the "This view does not exist" error page, which is quite nasty. 

This PR solves this problem in two steps: 

1. The login redirect is now a lot smarter and checks the menu first to find the next best accessible view instead
2. There's a new `home` option for User blueprints. This can be used to set a custom path or even use a string query to redirect to a page view, a file view, a user view or any other URL

### Simple paths
```yaml
title: Editor
home: /account
```

### Queries to panel views
```yaml
title: Editor
home: "{{ site.find('blog').children.first.panel.url }}" 
```

### Queries to pages
```yaml
title: Editor
home: "{{ site.find('blog').children.first.url }}" ​
```

## Todos

- [x] Check for valid URLs in home method
- [x] Remove any queries or other parts of the URL that could be used as an attack
- [x] Only allow URLs on the same domain

## Fixes

- https://github.com/getkirby/kirby/issues/3827

## Implements

- https://kirby.nolt.io/10

### CODE REVIEW

File: src/Panel/Panel.php [hunk 1]
```diff
@@ -153,11 +155,21 @@ public static function firewall(?User $user = null, ?string $areaId = null): boo
...
+    /**
+     * Handles redirection to the correct area
+     * after login or when visiting the login or installation
+     * views after being already authenticated
+     *
+     * @return void
+     */
+    public static function goHome(): void
+    {
+        static::go(Home::url());
```
Comment (distantnative) [inline src/Panel/Panel.php]: Do we need this new method? Now that we have the Home class, we could just put that code in the action callback below.

Comment (bastianallgeier) [reply src/Panel/Panel.php]: I was thinking about it as well. Could easily just be removed.

Comment (bastianallgeier) [reply src/Panel/Panel.php]: On the other hand, it's easier to test the results that way. The callback functions in routes are always a PITA to test.

Comment (distantnative) [reply src/Panel/Panel.php]: But what results? It's a whole method with one line inside that replaces a one line in the call-back function. In terms of test coverage, we'd still have to test now the function call to `goHome` in the callback function. Then we rather could test directly for `static::go(Home::url())`

Comment (bastianallgeier) [reply src/Panel/Panel.php]: @distantnative I changed it. It makes a lot more sense this way.

File: src/Panel/Home.php [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        $uri->query  = '';
+
+        // get a clean version of the URL
+        $url = $uri->toString();
+
+        // Don't further inspect URLs outside of the Panel
+        if (static::isPanelUrl($url) === false) {
+            return $url;
+        }
+
```
Comment (distantnative) [inline src/Panel/Home.php:242]: Shouldn't we already check here if the user has Panel access in general? And not in `static::alternative()`

Comment (bastianallgeier) [reply src/Panel/Home.php:242]: You are totally right. I looked at the entire class again after your comment and I think we can be a lot stricter with user checks in the other methods as well. I refactored it a bit and it's IMHO a lot cleaner now.

**Pull request merged**